tot/socket/socket.socketer.go

```diff
@@ -345,6 +351,7 @@ func (socket *Socket) listen() error {
 			socket.handleUnknown(response)
 		}
 
+		socket.listenMux.Lock()
 		if !socket.listening {
```

Comment:
Should break the loop by using `channel`
Example:

for {
     select {
            case <-quit:
            break
      default:
           // Continue loop
     }
}

Reply:
I agree, I'll make that change.